controllers/openstack_ansibleee_controller.go

```diff
@@ -407,11 +408,16 @@ func (r *OpenStackAnsibleEEReconciler) jobForOpenStackAnsibleEE(ctx context.Cont
 
 	// if we have any extra vars for ansible to use set them in the RUNNER_EXTRA_VARS
 	if len(instance.Spec.ExtraVars) > 0 {
+		keys := make([]string, 0, len(instance.Spec.ExtraVars))
+		for k := range instance.Spec.ExtraVars {
+			keys = append(keys, k)
+		}
+		sort.Strings(keys)
```

Comment:
I believe the real fix is to create a configMap with the extraVars and mount it in the right place: https://github.com/openstack-k8s-operators/edpm-ansible/blob/main/openstack_ansibleee/edpm_entrypoint.sh#L23

Reply:
the fix here LGTM. why would we need the ConfigMap?

Reply:
the reconcile is requeued while we wait for the job completion, so we sort it multiple times. I'm not sure about the configmap, but I think a better solution would avoid this sort on every run (but for not letting us blind with the CI failures, we can merge this and come up with a better solution later)